ratpack-core/src/main/java/ratpack/exec/internal/DefaultExecution.java

```diff
@@ -441,16 +462,34 @@ void enqueue(Block segment) {
 
     @Override
     void error(Throwable throwable) {
+      execStream = InErrorHandlerExecStream.INSTANCE;
+
       initial = null;
       if (segments != null) {
         segments.clear();
       }
-      try {
-        onError.execute(throwable);
-      } catch (Throwable errorHandlerError) {
-        LOGGER.error("error handler " + onError + " threw error (this execution will terminate):", errorHandlerError);
-        execStream = TerminalExecStream.INSTANCE;
-      }
+
+      final Execution currentExecution = DefaultExecution.this;
+      List<ExecutionErrorListener> errorListeners =
+        Lists.newArrayList(registry.getAll(ExecutionErrorListener.class));
+
+      controller.fork()
+        .onComplete(e -> {
```

Comment:
this allows us to pause the originating execution while the `ExecutionErrorListener` instances are working; then once they have completed, this will properly close off the original execution.

This is necessary to preserve the functionality that error handlers are invoked prior to the execution's `onComplete` handler.

Reply:
We should try to avoid this. We already had the behaviour of executing the error handlers before the on complete handler. What broke about that?